src/main/java/com/tanboonji/duke/Duke.java

```diff
@@ -0,0 +1,98 @@
+package com.tanboonji.duke;
+
+import java.util.Scanner;
+
+import com.tanboonji.duke.command.Command;
+import com.tanboonji.duke.command.HelpCommand;
+import com.tanboonji.duke.exception.DukeException;
+import com.tanboonji.duke.model.TaskList;
+import com.tanboonji.duke.parser.CommandParser;
+import com.tanboonji.duke.storage.Storage;
+import com.tanboonji.duke.ui.Ui;
+
+public class Duke {
+
+    private static final String FILE_DIR = "duke.txt";
+
+    private TaskList taskList;
+    private Ui ui;
+    private Storage storage;
+
+    private void run() {
+        storage = new Storage(FILE_DIR);
+        ui = new Ui();
+
+        try {
+            taskList = storage.load();
+        } catch (DukeException e) {
+            ui.print(e.getMessage());
+        }
+
+        ui.greet();
+
+        Scanner scanner = new Scanner(System.in);
+        boolean continueDuke = true;
+
+        while (continueDuke) {
+            String input = scanner.nextLine().trim();
+            continueDuke = processInput(input);
+        }
+    }
+
+    public void initialise() {
+        storage = new Storage(FILE_DIR);
+        ui = new Ui();
+
+        try {
+            taskList = storage.load();
+            ui.print("load successful");
+        } catch (DukeException e) {
+            ui.print(e.getMessage());
+        }
+
+        ui.greet();
+    }
+
+    private boolean processInput(String input) {
```

Comment:
Perhaps this method could be named in a way that makes it clearer that it returns a boolean?